From 3c24929eb7755bbd72cad4b18ad517922a4d21df Mon Sep 17 00:00:00 2001
From: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
Date: Thu, 15 Sep 2022 20:22:27 +0530
Subject: [PATCH] Skip phone Bluetooth profiles for Android Automotive

Bluetooth on/off toggling happens continuously.

Bluetooth adapter is trying to initialize phone
related Bluetooth profiles, which is not registered
for car service. So Bluetooth adapter turns off
after some times because it could not initialize those
profiles. Then it retries to turn on, again
it fails and this goes on.

Tracked-On: OAM-103692
Signed-off-by: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
---
 car_product/build/component-overrides.xml | 26 ++++++++++++++++-------
 1 file changed, 18 insertions(+), 8 deletions(-)

diff --git a/car_product/build/component-overrides.xml b/car_product/build/component-overrides.xml
index 510b2be7f..9eee2d2b2 100644
--- a/car_product/build/component-overrides.xml
+++ b/car_product/build/component-overrides.xml
@@ -28,29 +28,37 @@
     <component-override package="com.android.bluetooth" >
         <!-- Disable source profiles (typically used in phone) -->
         <!-- Controlled by profile_supported_a2dp -->
+<!--
         <component class=".a2dp.A2dpService" enabled="false" />
-
+-->
         <!-- Controlled by profile_supported_avrcp_target -->
+<!--
         <component class=".avrcp.AvrcpTargetService" enabled="false" />
-
+-->
         <!-- Controlled by profile_supported_hs_hfp -->
+<!--
         <component class=".hfp.HeadsetService" enabled="false" />
-
+-->
         <!-- Controlled by profile_supported_hid_device -->
+<!--
         <component class=".hid.HidDeviceService" enabled="false" />
-
+-->
         <!-- Controlled by profile_supported_hid_host -->
+<!--
         <component class=".hid.HidHostService" enabled="false" />
-
+-->
         <!-- Controlled by profile_supported_pbap -->
+<!--
         <component class=".pbap.BluetoothPbapActivity" enabled="false" />
         <component class=".pbap.BluetoothPbapService" enabled="false" />
-
+-->
         <!-- Controlled by profile_supported_map -->
+<!--
         <component class=".map.BluetoothMapService" enabled="false" />
         <component class=".map.BluetoothMapSettings" enabled="false" />
-
+-->
         <!-- Controlled by profile_supported_opp -->
+<!--
         <component class=".opp.BluetoothOppService" enabled="false" />
         <component class=".opp.BluetoothOppReceiver" enabled="false" />
         <component class=".opp.BluetoothOppLauncherActivity" enabled="false" />
@@ -59,9 +67,11 @@
         <component class=".opp.BluetoothOppIncomingFileConfirmActivity" enabled="false" />
         <component class=".opp.BluetoothOppTransferActivity" enabled="false" />
         <component class=".opp.BluetoothOppTransferHistory" enabled="false" />
-
+-->
         <!-- Controlled by profile_supported_sap -->
+<!--
         <component class=".sap.SapService" enabled="false" />
+-->
         <!-- End for disabling-->
 
         <!-- Enable sink profiles (typically used on a CarKitt) -->
-- 
2.17.1

